### PR TITLE
added enums and changed integers to be pointers to allow zeros be pas…

### PR DESCRIPTION
**Pull request by evgenigourvitch**
Hey,

added some enumerations which are defined in protocol + changed some other integers to pointers due to 0 values, which could be omitted from json.

thanks.

### CODE REVIEW

File: device.go [hunk 1]
```diff
@@ -209,7 +209,7 @@ type Device struct {
 	//   integer
 	// Description:
 	//   Network connection type. Refer to List 5.22.
-	Connectiontype int8 `json:"connectiontype,omitempty"`
+	Connectiontype *int8 `json:"connectiontype,omitempty"`
```
Comment (mxmCherry) [inline device.go]: Please, rename this field to `ConnectionType`, let's make it CamelCase. We'll release this as a next major version anyway, so renaming field is fine.

File: enums.go [hunk 2]
```diff
@@ -0,0 +1,282 @@
+package openrtb
```
Comment (mxmCherry) [inline enums.go]: Thank you for this, but, please, **remove it entirely**.

I won't accept enums into this repo yet. Either keep these enums in your own lib, or fork this repo and switch your code to use it.

I'll leave some style-critics on this file anyway, but don't fix them (they're more like "friendly advices") - just remove this file from PR.

File: enums.go [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+
+//enums and constants for tables 5.x.x (except 5.1 - IAB categories)
```
Comment (mxmCherry) [inline enums.go]: Check Go std lib - leaving space after `//` is more common.

So, `// comment` is cool, but `//comment` is not that cool.

File: enums.go [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+
+//5.2 Banner Ad Types
```
Comment (mxmCherry) [inline enums.go]: Stick comments to `const (...)` block opening, so it can be picked up by godoc.

File: enums.go [hunk 5]
```diff
@@ -0,0 +1,282 @@
...
+
+const (
+	EBannerAdTypeXHTMLTextAd int8 = iota + 1
```
Comment (mxmCherry) [inline enums.go]: Better not use this for OpenRTB enums - they're already defined in spec, don't rely on incremental `iota`.

Comment (mxmCherry) [reply enums.go]: Also, no need for `E` prefix. It won't really help you with autocompletion, so effectively it's useless.

Also, I haven't seen Hungarian notation in Go std lib.

And, as I remember, enum names don't really clash with object names, so really - no need for prefixing.

File: enums.go [hunk 6]
```diff
@@ -0,0 +1,282 @@
...
+	ENoBidReasonCodeBlockedPublisherOrSite
+	ENoBidReasonCodeUnmatchedUser
+	ENoBidReasonCodeDailyReaderCapMet
+	ENoBidReasonCodeDailyDomainCapMet
+)
+
+//5.25 Loss Reason Codes
+
+const (
+	ELossReasonCodesBidWon int8 = iota
```
Comment (mxmCherry) [inline enums.go]: So, you used `ENoBidReasonCode` in prev enum, but here you use `ELossReasonCodes` (plural). Naming should be consistent.

File: device.go (hunk 1 above)
Comment (evgenigourvitch) [reply device.go]: done.
thank you.

**Comment by mxmCherry** [comment]
Released as [v7.0.0](https://github.com/mxmCherry/openrtb/releases/tag/v7.0.0), thank you!

**Pull request merged**